cpp/src/prims/vertex_frontier.cuh

```diff
@@ -204,8 +203,6 @@ class key_bucket_t {
                                           merged_pair_first + merged_vertices.size())),
           handle_ptr_->get_stream());
         merged_tags.resize(merged_vertices.size(), handle_ptr_->get_stream());
-        merged_vertices.shrink_to_fit(handle_ptr_->get_stream());
-        merged_tags.shrink_to_fit(handle_ptr_->get_stream());
```

Comment:
Is there a reason we're not shrinking these here?

Reply:
`shrink_to_fit` basically allocates another array; copies data; and destroy the original array.

The copying part is expensive. If the expected number of duplicates is low, this really does not reduce memory footprint much, increase the peak memory usage (as we temporarily maintain two arrays), and increases execution time. Here two merged arrays typically don't have that many duplicates. I guess paying this cost for few % or less cut in memory is not a good trade-off.